src/components/elements/affiliate-date-picker.js

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import styled, { css } from 'styled-components'
+import DatePicker from 'react-date-picker'
+import moment from 'moment'
+import dayjs from 'dayjs'
+import device from 'themes/device'
+
+const DayPickerWrapper = styled.div`
+    width: 100%;
+    height: 100%;
+
+    .react-date-picker {
+        display: block;
+        font-size: var(--text-size-xs);
+        padding: 1rem 1rem 1rem 0.8rem;
+        height: 40px;
+    }
+    .react-date-picker__wrapper {
+        border: none;
+        border-radius: 5px;
+    }
+    .react-date-picker__calendar {
+        width: 280px;
+    }
+    .react-calendar__month-view__weekdays__weekday {
+        font-size: 15px;
+        font-weight: initial;
+    }
+    abbr {
+        border-bottom: none !important;
+        text-decoration: none !important;
+    }
+    .react-date-picker__inputGroup__input {
+        outline: none;
+    }
+    .react-date-picker__inputGroup__input:invalid {
+        background: none;
+    }
+    .react-calendar {
+        margin-top: 5px;
+        border: none;
+        border-radius: 5px;
+        box-shadow: 0 3px 6px -4px rgb(0 0 0 / 12%), 0 6px 16px 0 rgb(0 0 0 / 8%),
+            0 9px 28px 8px rgb(0 0 0 / 5%);
+    }
+    .react-calendar__navigation {
+        border-bottom: 1px solid #f3f4f5;
+    }
+    .react-calendar__navigation__arrow {
+        font-size: 30px;
+        font-weight: 100;
+    }
+    .react-calendar__navigation__arrow:hover {
+        background-color: none;
+    }
+    .react-calendar__tile--active,
+    .react-calendar__tile--active:hover,
+    .react-calendar__tile--hasActive,
+    .react-calendar__tile--hasActive:hover {
+        background-color: red;
+        border-radius: 5px;
+    }
+    label {
+        background-color: var(--color-white);
+        color: 'green';
+        transform: translate(-0.6rem, -2rem) scale(0.7);
+        ${({ isDateField, currentValue }) => {
+            return isDateField || currentValue
+                ? css`
+                      transform: translate(-0.6rem, -2.2rem) scale(0.7);
+                      color: var(
+                          --color-${({ error, labelFocusColor }) => (error ? 'red' : labelFocusColor)}
+                      );
+                  `
+                : css`
+                      transform: translate(0rem, 0rem) scale(1);
+                      color: var(--color-${({ error }) => (error ? 'red' : 'grey')});
+                  `
+        }}
+    }
+`
+const StyledLabel = styled.label`
+    /* stylelint-disable */
+    color: var(--color-${({ label_color }) => label_color || 'grey'});
+    /* stylelint-enable */
```

Comment:
```suggestion
    color: ${({ label_color }) => (label_color ? `var(--color-${label_color})` : 'var(--color-grey)')};
```
Don't need to disable stylelint here. You can do it this approach.